tests/contrib/test_multistore_file.py

```diff
@@ -133,6 +147,33 @@ def test_read_only_file_fail_lock(self):
             self.assertTrue(store._multistore._read_only)
         os.chmod(FILENAME, 0o600)
 
+    def test_read_only_file_fail_lock_no_warning(self):
+        filehandle, filename = tempfile.mkstemp('oauth2client_test.data')
+        os.chmod(filename, 0o400)
+
+        multistore = multistore_file._MultiStore(filename)
+
+        with mock.patch.object(multistore_file.logger, 'warn') as mock_warn:
+            multistore._warn_on_readonly = False
+            multistore._lock()
+            self.assertFalse(mock_warn.called)
```

Comment:
Officially I am supposed to recommended that you clean up the file after the test